tests/Microsoft.Identity.Test.Integration.netfx/SeleniumTests/InteractiveFlowTests.cs

```diff
@@ -229,23 +232,60 @@ private async Task<AuthenticationResult> RunTestForUserAsync(LabResponse labResp
                 .ExecuteAsync(new CancellationTokenSource(_interactiveAuthTimeout).Token)
                 .ConfigureAwait(false);
             userCacheAccess.AssertAccessCounts(2, 3);
+            AssertCCSRoutingInformationIsSent(factory, labResponse);
 
             account = await MsalAssert.AssertSingleAccountAsync(labResponse, pca, result).ConfigureAwait(false);
             userCacheAccess.AssertAccessCounts(3, 3);
             Assert.IsFalse(userCacheAccess.LastAfterAccessNotificationArgs.IsApplicationCache);
 
+            factory.RequestsAndResponses.Clear();
             Trace.WriteLine("Part 4 - Acquire a token silently");
             result = await pca
                 .AcquireTokenSilent(s_scopes, account)
                 .ExecuteAsync(CancellationToken.None)
                 .ConfigureAwait(false);
 
+            Trace.WriteLine("Part 5 - Acquire a token silently with force refresh");
+            result = await pca
+                .AcquireTokenSilent(s_scopes, account)
+                .WithForceRefresh(true)
+                .ExecuteAsync(CancellationToken.None)
+                .ConfigureAwait(false);
+
             await MsalAssert.AssertSingleAccountAsync(labResponse, pca, result).ConfigureAwait(false);
             Assert.IsFalse(userCacheAccess.LastAfterAccessNotificationArgs.IsApplicationCache);
+            AssertCCSRoutingInformationIsSent(factory, labResponse);
 
             return result;
         }
 
+        private void AssertCCSRoutingInformationIsSent(HttpSnifferClientFactory factory, LabResponse labResponse)
```

Comment:
This is fine, but we should also focus on unit testing. In particular, we don't have integrated tests for IWA.

Reply:
Do you know of a good way to validate headers in the Unit tests? The existing techniques that I used seem not to work with the mocks. @bgavrilMS

Reply:
There are 2 ways of doing this. The first way is to use the `ExpectedRequestHeaders` from the `MockHttpMessageHandler` object. 

The second way is to examine the HttpRequestMessage which the `MockHttpMessageHandler` exposes through the `ActualRequestMessage` property. It will have the actual headers sent.